lib/app/features/users/data/repositories/users_repository.dart

```diff
@@ -68,6 +71,34 @@ class UsersRepository implements IUsersRepository {
       return left(MapExceptionToFailure.map(error));
     }
   }
+
+  @override
+  Future<Either<Failure, ValidField>> report(
+      int clientId, String reason) async {
+    try {
+      const endPoint = '/report-profile';
+      final parameters = {'reason': reason, 'cliente_id': clientId.toString()};
+      final response = await _apiProvider!
+          .post(path: endPoint, parameters: parameters)
+          .parseBlockOrReport();
+      return right(response);
+    } catch (error) {
+      return left(MapExceptionToFailure.map(error));
```

Comment:
seria interessante a gente colocar no log, assim a gente consegue saber se por acaso acontecer algum problema durante o parse (mesma coisa para o método `block`)
```suggestion
    } catch (error, stack) {
      logError(error, stack);
      return left(MapExceptionToFailure.map(error));
```